prov/sockets/src/sock_msg.c

```diff
@@ -189,8 +189,14 @@ static ssize_t sock_ep_sendmsg(struct fid_ep *ep, const struct fi_msg *msg,
 		return -FI_EINVAL;
 
 	if (!tx_ctx->enabled)
-		return -FI_EOPBADSTATE;
-
+		return -FI_EOPBADSTATE;	
+#if ENABLE_DEBUG	
+	if(sock_ep->ep_type == FI_EP_DGRAM && sock_dgram_drop_rate > 0) {
+		atomic_inc(&sock_ep->domain->fab->num_send_msg);
+		if(!(atomic_get(&sock_ep->domain->fab->num_send_msg) % sock_dgram_drop_rate))
```

Comment:
For what this is trying to do, I don't think it matters if this is an atomic variable or not.